### PR TITLE
Model.boundingSphere was not properly scaling its center.

### PR DESCRIPTION
**Pull request by mramato**
When a scale is applied to a Model primitive, it scales the radius of `Model.boundingSphere` but not the center to match.  This is easy to see by using the Sandcastle 3D Models example and setting a `scale : 10` on the models and then switching to Cesium Man (the only model with a non-zero center). Instead of viewing the center of the model, as you get with scale of 1, you end up looking at his feet.

Also added a test to check for this, since we didn't have one.  The test passed in this branch but fails in master.

### CODE REVIEW

**Comment by mramato** [comment]
This may also be a solution to #1810, but I'm not sure.

File: Source/Scene/Model.js [hunk 1]
```diff
@@ -523,8 +524,11 @@ define([
                 }
                 //>>includeEnd('debug');
 
-                this._boundingSphere.radius = (this.scale * Matrix4.getMaximumScale(this.modelMatrix)) * this._initialRadius;
-                return this._boundingSphere;
+                var scale = (this.scale * Matrix4.getMaximumScale(this.modelMatrix));
+                var scaledBoundingSphere = this._scaledBoundingSphere;
+                scaledBoundingSphere.center = Cartesian3.multiplyByScalar(this._boundingSphere.center, scale, scaledBoundingSphere.center);
```
Comment (pjcozzi) [inline Source/Scene/Model.js]: I don't think we should use the scalar `scale` here.  We should use `Matrix4.getScale`.  Can discuss in person.

We should also do some testing with a `modelMatrix` with a scale and non-uniform scale.

**Comment by mramato** [comment]
Added support for non-uniform scales and another test case to verify it.  Ready for another look.

**Comment by pjcozzi** [comment]
Looks good.

**Pull request merged**